File: src/main/java/csorba/norberth/lab3/pieces/Position.java
package csorba.norberth.lab3.pieces;

public final class Position {
    public final int y;
    public final int x;

    public Position(int y, int x){
        this.y = y;
        this.x = x;
    }

    @Override
    public boolean equals(Object obj) {
        if(!(obj instanceof Position)){
            return false;
        }
        Position that = (Position) obj;
        return this.y == that.y && this.x == that.x;
    }

    @Override
    public int hashCode() {
        return y + x;
    }
}

File: Laboratoare_gbir2109/DP_Lab3_Homework/DP_Lab3_Homework/Pieces/Position.cs
namespace DP_Lab3_Homework.Pieces
{
    public class Position
    {
        public int HorizontalPos { get; set; }

        public int VerticalPos { get; set; }

        public override bool Equals(object obj)
        {
            if (this == obj) return true;
            if (obj == null || this.GetType() != obj.GetType()) return false;
            Position position = (Position)obj;
            return VerticalPos == position.VerticalPos &&
                    HorizontalPos == position.HorizontalPos;
        }
    }
}